messagebus/producer.py

```diff
@@ -0,0 +1,336 @@
+        :param record: record/message to be written
+        :param on_delivery_callback (callable(KafkaError, Message), optional): Delivery
+            report callback to call (from
+            :py:func:`SerializingProducer.poll` or
+            :py:func:`SerializingProducer.flush` on successful or
+            failed delivery.
+        :return:
+        """
+        self.log_debug("Producing records to topic {}.".format(topic))
+        try:
+            # The message passed to the delivery callback will already be serialized.
+            # To aid in debugging we provide the original object to the delivery callback.
+            self.producer.produce(
+                topic=topic,
+                key=key,
+                value=value,
+                headers=self.__message_header_generator(),
+                on_delivery=self.delivery_report,
+            )
+            # Serve on_delivery callbacks from previous asynchronous produce()
+            self.producer.poll(0)
+            return True
+        except ValueError as ex:
+            traceback.print_exc()
+            self.log_error("Invalid input, discarding record...{}".format(ex))
+        return False
+
+    def produce_sync(self, topic: str, value, key="default") -> bool:
+        """
+        Produce records for a specific topic
+        :param topic: topic to which messages are written to
+        :param record: record/message to be written
+        :return:
+        """
+        try:
+            self.log_debug("Record type={}".format(type(value)))
+            self.log_debug("Producing key {} to topic {}.".format(key, topic))
+            self.log_debug("Producing record {} to topic {}.".format(value, topic))
+
+            # Pass the message synchronously
+            self.producer.produce(
+                topic=topic,
+                key=key,
+                value=value,
+                headers=self.__message_header_generator(),
+            )
+            self.producer.flush()
+            return True
+        except ValueError as ex:
+            self.log_error("Invalid input, discarding record...")
+            self.log_error(f"Exception occurred {ex}")
+            self.log_error(traceback.format_exc())
+        except Exception as ex:
+            self.log_error(f"Exception occurred {ex}")
+            self.log_error(traceback.format_exc())
+        return False
+
+
+if __name__ == "__main__":
+    import sys
+    import time
+    from messagebus.admin import AdminApi
+    from confluent_kafka.schema_registry import record_subject_name_strategy
+
+    if len(sys.argv) != 5:
+        sys.stderr.write(
+            "Usage: python %s <bootstrap-brokers> <schema-registry-url> <username> <password>\n"
+            % sys.argv[0]
+        )
+        sys.exit(1)
+
+    broker = sys.argv[1]
+    schema_registry_url = sys.argv[2]
+    username = sys.argv[3]
+    password = sys.argv[4]
+
+    # conf init
+    test_conf = {
```

Comment:
@goFrendiAsgard @sulthonzh what is python best practice for test code? 
should it be placed on different files?

Reply:
I think so, this one should be located on `producer_test.py` or something. WDYT @sulthonzh?
I will recommend using pytest for this

Reply:
https://github.com/kata-ai/messagebus-kafka-python/pull/1/commits/c1d294dad74f47b4beea2ad0a3b1c044c5e6419e